python/src/com/jetbrains/python/psi/impl/PythonLanguageLevelPusher.java

```diff
@@ -166,11 +173,22 @@ public boolean acceptsDirectory(@NotNull VirtualFile file, @NotNull Project proj
 
   private static final FileAttribute PERSISTENCE = new FileAttribute("python_language_level_persistence", 2, true);
 
+  private static boolean areLanguageLevelsCompatible(int oldLevelOrdinal, int newLevelOrdinal) {
+    if (oldLevelOrdinal == -1 || newLevelOrdinal == -1) {
+      return false;
+    }
+    int minLevel = Math.min(oldLevelOrdinal, newLevelOrdinal);
+    int maxLevel = Math.max(oldLevelOrdinal, newLevelOrdinal);
+
+    return COMPATIBLE_LEVELS.contains(Pair.create(minLevel, maxLevel));
+  }
+
   public void persistAttribute(@NotNull Project project, @NotNull VirtualFile fileOrDir, @NotNull LanguageLevel level) throws IOException {
     final DataInputStream iStream = PERSISTENCE.readAttribute(fileOrDir);
+    int oldLevelOrdinal = -1;
     if (iStream != null) {
       try {
-        final int oldLevelOrdinal = DataInputOutputUtil.readINT(iStream);
+        oldLevelOrdinal = DataInputOutputUtil.readINT(iStream);
```

Comment:
Let's convert int value to enum, or null if there is no enum value with such ordinal